tests/kernel/mem_protect/stackprot/testcase.yaml

```diff
@@ -1,4 +1,9 @@
 tests:
   kernel.memory_protection.stackprot:
-    arch_exclude: nios2 xtensa posix
+    arch_exclude: nios2 xtensa posix riscv32
     tags: kernel ignore_faults userspace
+  kernel.memory_protection.stackprot_riscv:
+    arch_allow: riscv32
+    tags: kernel ignore_faults userspace
+    extra_configs:
+      - CONFIG_KOBJECT_TEXT_AREA=512
```

Comment:
this test shouldn't require this modification. why do you need to change this here?

Reply:
Is it due to canary feature.
I remove this patch and fix this like that:
`default 512 if STACK_CANARIES && RISCV`
(change pushed)